### PR TITLE
SW-3722 Track subzone finished-planting time

### PR DESCRIPTION
**Pull request by sgrimm**
Some of the calculated observation results are conditional on whether or not all
the subzones in a zone or at a site are finished planting. The condition needs to
test whether the subzones were finished planting at the time of the observation,
not at the current time, or we'll calculate incorrect values for observations that
happened before a site was finished planting.

Replace the "finished planting" flag on subzones with a timestamp. The subzone API
payloads still include the boolean value, so this change should be transparent to
clients, except that the boolean field is no longer present in the search API.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1193** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1193" target="_blank">Graphite</a>  👈
    * **PR #1183** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1183" target="_blank">Graphite</a> 
      * **PR #1194** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1194" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1193?utm_source=stack-comment).

File: src/main/kotlin/com/terraformation/backend/tracking/api/PlantingSubzonesController.kt [hunk 1]
```diff
@@ -61,5 +62,6 @@ data class ListPlantingSubzoneSpeciesResponsePayload(
 data class UpdatePlantingSubzoneRequestPayload(
     val finishedPlanting: Boolean,
 ) {
-  fun applyTo(row: PlantingSubzonesRow) = row.copy(finishedPlanting = finishedPlanting)
+  fun applyTo(row: PlantingSubzonesRow): PlantingSubzonesRow =
+      row.copy(finishedTime = if (finishedPlanting) Instant.EPOCH else null)
```
Comment (karthikbtf) [inline src/main/kotlin/com/terraformation/backend/tracking/api/PlantingSubzonesController.kt]: Add a comment why Instant.EPOCH? I understand it is used as a non-null flag/check for updatePlantingSubzone.

File: src/main/kotlin/com/terraformation/backend/search/table/PlantingSubzonesTable.kt [hunk 2]
```diff
@@ -42,7 +42,7 @@ class PlantingSubzonesTable(tables: SearchTables) : SearchTable() {
       listOf(
           geometryField("boundary", PLANTING_SUBZONES.BOUNDARY),
           timestampField("createdTime", PLANTING_SUBZONES.CREATED_TIME, nullable = false),
-          booleanField("finishedPlanting", PLANTING_SUBZONES.FINISHED_PLANTING, nullable = false),
+          timestampField("finishedTime", PLANTING_SUBZONES.FINISHED_TIME, nullable = false),
```
Comment (karthikbtf) [inline src/main/kotlin/com/terraformation/backend/search/table/PlantingSubzonesTable.kt]: i don't want to trigger a sweeping change :( but thoughts on `finishedPlantingTime`?

**Review by karthikbtf — APPROVED** [review]
Looks good to me 👍

File: src/main/kotlin/com/terraformation/backend/search/table/PlantingSubzonesTable.kt (hunk 2 above)
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/search/table/PlantingSubzonesTable.kt]: Yeah, that's clearer, agreed.

**Review by karthikbtf — APPROVED** [review]
Thank you!!

**Comment by sgrimm** [comment]
@sgrimm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1193).

**Comment by sgrimm** [comment]
@sgrimm merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1193).

**Pull request merged**